src/games/strategy/engine/ClientFileSystemHelper.java

```diff
@@ -0,0 +1,149 @@
+package games.strategy.engine;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URL;
+import java.net.URLDecoder;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import games.strategy.debug.ClientLogger;
+import games.strategy.engine.framework.GameRunner;
+import games.strategy.engine.framework.GameRunner2;
+import games.strategy.util.Version;
+
+/**
+ * Pure utility class, final and private constructor to enforce this
+ * WARNING: do not call ClientContext.getInstance() in this class. ClientContext call this class in turn
+ * during construction, depending upon ordering this can cause an infinite call loop.
+ */
+public final class ClientFileSystemHelper {
+  private ClientFileSystemHelper() {
+
+  }
+
+  /** This method is available via ClientContext */
+  public static File getRootFolder() {
+    final String fileName = getGameRunnerFileLocation("GameRunner2.class");
+
+    final String tripleaJarName = "triplea.jar!";
+    if (fileName.contains(tripleaJarName)) {
+      return getRootFolderRelativeToJar(fileName, tripleaJarName);
+    }
+
+//    final String tripleaJarNameWithEngineVersion = getTripleaJarWithEngineVersionStringPath();
+    if (fileName.contains("triplea_") && fileName.contains(".jar!")) {
```

Comment:
Yet another update, looks like calling 'getRootFolder()' really was just not needed, so set this back just in case..